docs/developer/decision-records/2024-10-09-vault-authentication-refactor/README.md

```diff
@@ -0,0 +1,72 @@
+# Decision Record Refactor of the HashiCorp Vault extension
+
+## Decision
+
+Refactor the HashiCorp Vault extension and make it extensible with different methods of authentication.
+
+## Rationale
+
+The current implementation of the authentication in the HashiCorp Vault extension has no way to add new authentication methods and to switch between them.
+It goes against the EDC extensibility model and needs to be remedied.
+Additionally, extracting the current implementation of the token authentication into its own extension will improve readability and maintainability of the code.
+
+## Affected Areas
+
+HashiCorp Vault extension
+
+## Approach
+
+To allow an extensible authentication module for HashiCorp Vault, an implementation similar to the existing management API authentication is possible.
```

Comment:
> an implementation similar to the existing management API authentication is possible

if we want to state this, then there should be some sort of comparison, but is missing from the Approach.
I would just state that we will follow the registry pattern.

for completeness: please note that there's not management specific authentication, it's, in fact, usable also in other api contexts